internal/dangling/dangling.go

```diff
@@ -127,3 +122,18 @@ func isOwnedBy(element metav1.Object, owner metav1.Object) bool {
 	}
 	return false
 }
+
+func buildDaemonSetNames(instance *nropv1.NUMAResourcesOperator, trees []nodegroupv1.Tree) sets.Set[string] {
+	expectedDaemonSetNames := sets.New[string]()
+	for _, tree := range trees {
+		poolName := tree.NodeGroup.PoolName
+		if poolName != nil && *poolName != "" {
```

Comment:
no need to validate it's not empty string, we do that early at the validation phase for node group

Reply:
I see.
As a general comment I would say that any contributor, that is not too familiar with the project and follow best practices, would add this trivial check since this is a pointer which is given to the function from unknown source (unknown in the context of the function per se)

Reply:
> I see. As a general comment I would say that any contributor, that is not too familiar with the project and follow best practices, would add this trivial check since this is a pointer which is given to the function from unknown source (unknown in the context of the function per se)

Not fully onboard (but not a problem in this case). I'd prefer adding testcases covering the flows first, which would help highlitghting cases like that. Reviews should also highlight this more.
Surely we should (my fault here) better docs/architecture docs/developer README to outline general guidelines like that.